include/lsst/afw/detection/FootprintSet.h

```diff
@@ -67,6 +68,21 @@ class FootprintSet {
     FootprintSet(image::Image<ImagePixelT> const& img, Threshold const& threshold, int const npixMin = 1,
                  bool const setPeaks = true);
 
+    /**
+     * Find a FootprintSet given an Image and a threshold, specifying the schema for the peak records.
+     *
+     * This is used to merge FootprintSets while preserving the input peak schema.
+     *
+     * @param img Image to search for objects
+     * @param threshold threshold to find objects
+     * @param npixMin minimum number of pixels in an object
+     * @param setPeaks should I set the Peaks list?
+     * @param peakSchema Schema for peak records, even if we don't measure them here.
+     */
+    template <typename ImagePixelT>
+    FootprintSet(image::Image<ImagePixelT> const& img, Threshold const& threshold,
+                 table::Schema const& peakSchema, int const npixMin = 1, bool const setPeaks = true);
```

Comment:
It should also be possible to write this as a defaulting version of the existing similar overload by moving the `peakSchema` argument to the end:
```
template <typename ImagePixelT>
FootprintSet(image::Image<ImagePixelT> const& img, Threshold const& threshold,
             int const npixMin = 1, bool const setPeaks,
             table::Schema const& peakSchema = PeakTable::makeMinimalSchema())
```
In C++ that default value is effectively stored as a `static` variable, and it'll be copied into each call, so (unlike Python) it should be safe despite `Schema` being mutable.